src/plugins/discover/public/context_awareness/profile_providers/log_document_profile/profile.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createRegExpPatternFrom } from '@kbn/data-view-utils';
+import { DataTableRecord } from '@kbn/discover-utils';
+import { DocumentProfileProvider, DocumentType } from '../../profiles';
+
+export const ALLOWED_LOG_DOCUMENT_INDICES = createRegExpPatternFrom([
+  'logs',
+  'auditbeat',
+  'filebeat',
+  'winlogbeat',
+]);
```

Comment:
@tonyghiani would you agree that this might be something we want to share between the two profiles?

Reply:
We surely want to share this for both data source and document resolution, let's find a place where we can make easily accessible this and also fill in a possible user configuration setting.

Reply:
As this will most likely depend on the `uiSettings`, I'm considering it might be worth passing the core/plugins to the `resolve` methods, in case the resolution steps become more complex in future.

For this case, it might be an idea also to craft a small service (e.g. `contextAwareness.isLogsIndexPattern(index: string)`) which uses this hard-coded set of patterns, but that can rely on the uiSettings or any other deps if necessary.
What I mean is having something sort of:
```ts
resolve: ({ record, services }) => {
  const isLogRecord =
    getDataStreamType(record).includes('logs') ||
    hasFieldsWithPrefix('log.')(record) ||
    getIndices(record).some(services.contextAwarness.isLogsIndexPattern)

  if (!isLogRecord) {
    return { isMatch: false };
  }

  return {
    isMatch: true,
    context: {
      type: DocumentType.Log,
    },
  };
};
```
Wdyt? Let me know if there is any other option you would prefer!